<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="resolution" default="VGA"/>
  <arg name="fps" default="30"/>
  <arg name="exposure" default="100"/>
  <arg name="gain" default="0"/>
  <arg name="auto_exposure" default="0"/>

  <node pkg="snap_cam" type="snap_cam_publisher_node" name="snap_cam_optflow_publisher" output="screen">

    <!--
    "optflow" or "highres" -->
    <param name="camera_type" value="optflow"/>

    <!--
    optical flow: QVGA : (320,240)
                  VGA  : (640,480) -->
    <param name="camera_resolution" value="$(arg resolution)"/>

    <!--
    optical flow: 15, 24, 30, 60, 90 -->
    <param name="camera_fps" value="$(arg fps)"/>

    <!--
    optical flow: integer between 0 and 511 -->
    <param name="exposure" value="$(arg exposure)"/>

    <!--
    optical flow: integer between 0 and 255 -->
    <param name="gain" value="$(arg gain)"/>

    <!--
    optical flow: auto exposure (bool) -->
    <param name="auto_exposure" value="$(arg auto_exposure)"/>

    <param name="topic_name" value="image"/>
  </node>
</launch>
